platform/viewer/src/connectedComponents/Viewer.js

```diff
@@ -197,18 +197,38 @@ class Viewer extends Component {
         ]);
       }
 
+      const activeViewport = this.props.viewports[this.props.activeViewportIndex];
+      const activeDisplaySetInstanceUID =
+        activeViewport ? activeViewport.displaySetInstanceUID : undefined;
       this.setState({
-        thumbnails: _mapStudiesToThumbnails(studies),
+        thumbnails: _mapStudiesToThumbnails(studies, activeDisplaySetInstanceUID),
       });
     }
   }
 
   componentDidUpdate(prevProps) {
-    const { studies, isStudyLoaded } = this.props;
+    const {
+      studies,
+      isStudyLoaded,
+      activeViewportIndex,
+      viewports
+    } = this.props;
+
+    const activeViewport = viewports[activeViewportIndex];
+    const activeDisplaySetInstanceUID =
+      activeViewport ? activeViewport.displaySetInstanceUID : undefined;
```

Comment:
This change is really necessary? I would try to avoid changes to viewer.js lifecycle hooks. This function is kinda delicate and could potentially decrease the viewer performance or introduce bugs.

If this is necessary for it to work, can you describe why? maybe we can find another alternatives

Reply:
this is necessary since when chaning active viewport or displayset in the current active viewport, we have to recalculate the 'active' property for the thumbnail (i.e. if the border of the thumbnail images has to be highlithed). 

I run perfomance profiles and and I didn't see any change in the perfomance.

I also measured _mapStudiesToThumbnails  
1) the first run (at loading time): some series ~20 millisec, all the other < 1 microsec to run
2) successive runs (i.e. when clicking to change active displayset) : all series < 1 microsec to run

Notice that _mapStudiesToThumbnails essentially runs two functions: 
A) _checkForSeriesInconsistencesWarnings
B) _isDisplaySetActive 

(B) is the one that we care to rerun to compute the 'active' property
(A) Is computed only once (the first time at loading time), because we cache the values at the first time.

So I don't think we will have perfomances issues, I tested also on several other datasets and I did not find issues.